ios/Surveys/Sources/Presentation/Modules/Home/HomeViewModel.swift

```diff
@@ -22,8 +22,22 @@ final class HomeViewModel: ObservableObject {
     @Injected(Container.dateTime) private var dateTime: DateTimeProtocol
     @Injected(Container.dateTimeFormatter) private var dateTimeFormatter: DateTimeFormatterProtocol
 
+    private var currentPage: Int = 0
+    private var surveys: [Survey] = []
+
     private var bag = Set<AnyCancellable>()
 
+    var surveyDetailViewModel: SurveyDetailViewModel {
+        let survey = surveys[currentPage]
+        let surveyArgument = SurveyArgument(
+            id: survey.id,
+            title: survey.title,
+            description: survey.description_,
+            imageURLString: survey.coverImageUrl
+        )
+        return SurveyDetailViewModel(survey: surveyArgument)
+    }
```

Comment:
I might not be familiar with the ios project template. However, from the system design, does it make sense to build the new SurveyDetailViewModel from this HomeViewModel? Does it make sense to make them constraint together at least from the import 💭

Reply:
@luongvo Currently, we are creating the next screen's view model from the previous screen's view model. The idea is more supportive of connecting between screens in this case. We will use something like this
```swift
let nextViewModel = NextViewModel()
nextViewModel.delegate = self
return nextViewModel
```

Moreover, in iOS, we also creating different view models for subviews as well: You can check this: https://github.com/nimblehq/mvvm-rxswift-demo/blob/develop/MVVMRxSwiftDemo/Sources/Presentation/Modules/Home/HomeViewModel.swift#L42

>Does it make sense to make them constraint together at least from the import 💭
I'm not sure I get this point, can you give me more detail?

Reply:
@suho Thanks for your explanation. Good to know the different approach in the iOS platform 🤗